Sources/SwiftMemcache/MemcachedResponseDecoder.swift

```diff
@@ -151,6 +166,28 @@ struct MemcachedResponseDecoder: NIOSingleStepByteToMessageDecoder {
             let response = MemcachedResponse(returnCode: returnCode, dataLength: dataLength)
             self.nextStep = .returnCode
             return .returnDecodedResponse(response)
+
+        case .decodeValue(let returnCode, let dataLength, let flags):
+            guard buffer.readableBytes >= dataLength else {
+                return .waitForMoreBytes
+            }
+
+            var data: ByteBuffer = .init()
+            for _ in 0..<dataLength {
+                let byte = buffer.readInteger(as: UInt8.self)
+                data.writeInteger(byte!)
+            }
+
+            guard let nextByte = buffer.readInteger(as: UInt8.self),
+                  nextByte == UInt8.carriageReturn,
+                  let nextNextByte = buffer.readInteger(as: UInt8.self),
+                  nextNextByte == UInt8.newline else {
+                return .waitForMoreBytes
```

Comment:
When we change the logic here to guard that we have these bytes readable as well we should probably `fatalError` here because it should never happen